pkg/fetcher/fetcher.go

```diff
@@ -280,13 +296,32 @@ func (u *URLFetcher) fetch(ctx context.Context, url *url.URL, reqHdrs *http.Head
 			}
 		}
 
-		err = fmt.Errorf("Nonretryable error '%s (%d)': URL %s.", http.StatusText(u.StatusCode), u.StatusCode, url)
+		// for all other non-retryable client error, grab the error message if there is one (#5951)
+		msg, err := u.extractErrResponseMessage(res.Body)
+		if err != nil {
+			log.Debugf("Nonretryable client error in unconventional format: %s", err)
+			err = fmt.Errorf("Nonretryable error '%s (%d)': URL %s", http.StatusText(u.StatusCode), u.StatusCode, url)
```

Comment:
I think this is too verbose.  Users don't like to see to information overload.  If we're returning error, we don't really need to say 'nonretryable error'.  It's redundant, and they have no idea what that means.  It's fine for our logs.